MouseUnSnag/ScreenHandling/DisplayList.cs

```diff
@@ -67,32 +67,42 @@ public Display WhichScreen(Point P)
         }
 
         /// <summary>
-        /// Find the first monitor (first one we come across in the for() loop)
-        /// that is in the direction of the point.
+        /// Find the all the monitors that are in the direction of the point. Since
+        /// this method is used for monitor jumping, diagonal monitors are included in
+        /// horizontal and vertical directions, and horizontal and vertical monitors
+        /// are included in diagonal directions.
         /// </summary>
         /// <param name="dir">Direction</param>
         /// <param name="curScreen">Current Screen</param>
         /// <returns></returns>
-        public Display ScreenInDirection(Point dir, Rectangle curScreen)
+        private IEnumerable<Display> ScreensInDirection(Point dir, Rectangle curScreen) => All.Where(screen => {
+            var screenDir = GeometryUtil.OutsideDirection(curScreen, screen.Bounds);
+            return screenDir.X == dir.X && Math.Abs(screenDir.Y - dir.Y) <= 1.0
+                || screenDir.Y == dir.Y && Math.Abs(screenDir.X - dir.X) <= 1.0;
+        });
+
+        /// <summary>
+        /// Finds the best screen to "jump" the cursor. The best screen to jump to should be in
+        /// the direction the cursor is moving from the current screen, and should match human
+        /// intuition. Currently, we pick the closest screen by euclidean distance.
+        /// </summary>
+        /// <param name="mouse">Mouse</param>
+        /// <param name="curScreen">Current Screen</param>
+        /// <returns></returns>
+        public Display JumpScreen(Point mouse, Rectangle curScreen)
         {
-            // Screen must be strictly above/below/beside. For instance, for a monitor to be
-            // "above", the monitor's Bottom equal to the current screen's Top ("current
-            // screen" is where the Cursor (NOT the mouse!!) is currently).
-            foreach (var s in All)
+            var candidates = ScreensInDirection(GeometryUtil.OutsideDirection(curScreen, mouse), curScreen);
```

Comment:
Running this code, I notice that the cursor is having difficulty moving along the top of a display where there is not another display above it. The debug statements reveal a lot of 'jumps' of the mouse to the `curScreen`:

```1183: StuckDirection/Distance{X=0,Y=-1}/{X=0,Y=-1} cur_mouse:{X=2559,Y=-1}  prev_mouse:{X=2565,Y=-1} ==? cursor:{X=2560,Y=0} (OnMon#2/)  #UnSnags 1179   --STUCK--   
  > JumpScreen
{X=2560,Y=0} -> {X=2560,Y=0}
1184: StuckDirection/Distance{X=0,Y=-1}/{X=0,Y=-1} cur_mouse:{X=2554,Y=-1}  prev_mouse:{X=2559,Y=-1} ==? cursor:{X=2555,Y=0} (OnMon#2/)  #UnSnags 1180   --STUCK--   
  > JumpScreen
{X=2555,Y=0} -> {X=2555,Y=0}
1185: StuckDirection/Distance{X=0,Y=-1}/{X=0,Y=-1} cur_mouse:{X=2552,Y=-1}  prev_mouse:{X=2554,Y=-1} ==? cursor:{X=2553,Y=0} (OnMon#2/)  #UnSnags 1181   --STUCK--   
  > JumpScreen
{X=2553,Y=0} -> {X=2553,Y=0}
```

I think because `curScreen` itself is being considered as a `candidate` for jumping the cursor?

Reply:
Thanks for testing a build of this. Yep, there's a bug in ScreensInDirection.

```c#
return screenDir.X == dir.X && Math.Abs(screenDir.Y - dir.Y) <= 1.0
    || screenDir.Y == dir.Y && Math.Abs(screenDir.X - dir.X) <= 1.0;
```

should be

```c#
return screenDir.X != 0 && screenDir.X == dir.X && Math.Abs(screenDir.Y - dir.Y) <= 1
    || screenDir.Y !- 0 && screenDir.Y == dir.Y && Math.Abs(screenDir.X - dir.X) <= 1;
```